Better `install.bat` file

KAuser2094 opened the pull request:
# Windows `install.bat` File Change
- Now uses batch functions/labels (mostly  an aesthetic change)
- Now allows for different install options
- Can create a `.env` file in the appropriate directory
- Can optionall create the `.yolo-safety-off` and `.openai.apikey` files in the `~` directory if chosen
## Installation
### Default
When `install.bat` is ran it will open a terminal (if not already open)
> Do you want to install with default settings (or (C)ancel) (or create (O)ptional files)? <Y/N/C/O>:

Choosing `Y` will proceed with installing yolo as it would have done previously

### ...Not-Default
Choosing `N` will proceed to give you options in installing.
#### Using Created Directory or Git Repository
>Do you want to install a seperate 'yolo-ai-cmdbot\' directory (otherwise will use this repository folder)? <Y/N>:

The extra folder `yolo-ai-cmdbot\` folder made is somewhat redundant if you plan to keep the git repository so added an option to simply forgo it and use the directory of the cloned git repository instead. The `yolo.bat` file created will then run the `yolo.py` inside the local git folder instead.
#### Choosing File Paths
If `Y` is chosen to install a seperate 'yolo-ai-cmdbot\' directory
>Currently 'yolo-ai-cmdbot\' will be made in 'C:\Users\kause\' . Is this OK? <Y/N>: N
Enter a path for 'yolo-ai-cmdbot\' to be made:C:\Users\kause\Downloads
Currently 'yolo-ai-cmdbot\' will be made in 'C:\Users\kause\Downloads' . Is this OK? <Y/N>:

>Currently 'yolo.bat' will be made in 'C:\Users\kause\' . Is this OK? <Y/N>: N
Enter a path for 'yolo.bat' to be made:C:\Users\kause\Downloads\Commands\ScriptPath
Currently 'yolo.bat' will be made in 'C:\Users\kause\Downloads\Commands\ScriptPath' . Is this OK? <Y/N>:

Should be self explanatory, but the user will be able to choose a directory for the created 'yolo-ai-cmdbot\' and `yolo.bat`. There isn't a check to see if the filepath exists because I already feel as if this was overengineered.

### Installation message
>Installing Yolo (Using Created Directory)
Yolo Directory:
Installing to C:\Users\kause\Downloads\yolo-ai-cmdbot\
        1 file(s) copied.
        1 file(s) copied.
Yolo Batch (Directory):
yolo.py should be in C:\Users\kause\Downloads\yolo-ai-cmdbot\...
Found: Creating "yolo.bat" in C:\Users\kause\Downloads\Commands\ScriptPath
        1 file(s) copied.
Yolo .Env:
Creating ".env" (if not already exists) in C:\Users\kause\Downloads\yolo-ai-cmdbot\

Doesn't matter.

### Optional Files
After installation (or choosing `O` to the first choice) you will be able to choose to create a `.openai.apikey` and/or `.yolo-safety-off` file(s) in your `~` directory
>Do you want to make a '.openai.apikey' file at C:\Users\kause\ to hold your apikey? <Y/N>: 

>Do you want to make a '.yolo-safety-off' file at C:\Users\kause\ to disable safety? <Y/N>:

>Yolo OpenAi ApiKey:
Creating ".open.apikey" (if not already exists) in C:\Users\kause\
        1 file(s) copied.

>Yolo Safety Off:
Creating ".yolo-safety-off" in C:\Users\kause\
        1 file(s) copied.

Self explanatory

### Guide message
After an installation (so not picking option `O` on the first choice) a message will be shown in the terminal.
>Finished installing yolo-ai-cmdbot...
You can run commands by being in the same directory as your "yolo.bat" file ( It is in C:\Users\kause\Downloads\Commands\ScriptPath ) with the following command:
".\yolo.bat <prompt>"
You can also put your "yolo.bat" file into a $PATH directory and run it like so, instead:
"yolo <prompt>"
You should have "C:\Windows\System32" as a path in the $PATH environment variable but you can write the command "echo $env:PATH" into PowerShell for more options...

# General
Changed `README.md` to reflect changes to usage of `install.bat`

wunderwuzzi23 commented on install.bat:
```diff
@@ -1,25 +1,164 @@
+set "installing=1"
+
+:: Set Variables To User Defined If Needed 
+:: (This was as painful to make as it looks)
```
:) Thanks for building this.

wunderwuzzi23 commented on install.bat:
```diff
@@ -1,25 +1,164 @@
+if /i %createDIR%==1 ( call :install_yolo_directory ) else ( call :install_yolo_repository )
+
+::Optional files:
+:choice_default_4
```
Is it possible to skip this step if the user does a default install? 

That way the default is a quick install, and the user won't be faced with the decision about the safety mode. A first time user might not know what the safety mode means and accidentally create the file.

wunderwuzzi23 commented on install.bat:
```diff
@@ -1,25 +1,164 @@
+:choice_default_4
+choice /n /c YN /m "Do you want to make a '.openai.apikey' file at %HOMEDRIVE%%HOMEPATH%\ to hold your apikey? <Y/N>:"
+set /a createAPIKEY=!ERRORLEVEL!
+choice /n /c YN /m "Do you want to make a '.yolo-safety-off' file at %HOMEDRIVE%%HOMEPATH%\ to disable safety? <Y/N>:"
```
For the same reason about safety and user's accidentally turning it off, maybe add the following text to the question: "If you don't know what this means, select No".

wunderwuzzi23 commented:
Thanks so much! This is very quite advanced now. 👍 

I added a few comments to the code (mostly about the creation of the "safety off" feature) in default install. Let me know if those change ideas sound good to you? 

Again, thanks for spending the time to improving this project.

KAuser2094 commented:
# Addressing Comments:
## Better Messages
Created more `:print_[type]` functions which holds messages. 
These include:
- An introduction message
- Some more guide messages (about api key and warnings about moving files) 
- Better messages for all the choices (With recommended inputs where I thought appropriate)
## Optional Files
Seperated the choice to and creation of `.yolo-safety-off` and `.openai.apikey` to a seperate function that won't be called on default install.
## Other
### Error Catching
Added some choice labels in case or ERRORLEVEL being wrong (example, it is 0 if user ^C the terminal)
### File Path Checking
Went back on my opinion that this was over engineered, might as well add it. It didn't even add to the lines of code.
### Actually checked `export` terminal command
Never actually checked if it works on Windows. It doesn't. :) 
So added commands that actually do work to the guide message (As in "Better Messages") and to the `README.md`
### Change `README.md`
Obviously, had to change `README.md`
